### ISSUE TITLE
Edit item in collection ? 

### ISSUE THREAD

**Issue by RoxDevvv** [nylo-core/nylo#106]
edit item in collection ? 
await NyStorage.addToCollection("a_storage_key", newItem: "1");

**Comment by RoxDevvv** [comment]
maybe something builtin like that will be good
```
static Future updateInCollection<T>(int index, T newItem,
    {required String key}) async {
  List<T> collection = await readCollection<T>(key);
  if (collection.isEmpty || index < 0 || index >= collection.length) return;

  collection[index] = newItem;
  await saveCollection<T>(key, collection);
}
```

**Comment by agordn52** [comment]
Hi @RoxDevvv,

This is a cool idea!

I'll work on it later today and hopefully push it up for tomorrow 🚀

**Comment by agordn52** [comment]
Hi @RoxDevvv,

`updateCollectionByIndex` now available in <b>nylo_framework</b> [v5.20.9](https://pub.dev/packages/nylo_framework) 🔥

Here's an example of how to use it.

``` dart
await NyStorage.updateCollectionByIndex(2, (item) {
      return 'New value';
}, key: 'my_storage_key');


// Example on a Model
await NyStorage.updateCollectionByIndex<User>(5, (User item) {
  item.name = "User Updated";
  item.email = "updated@aol.com";
  return item;
}, key: 'my_storage_key');
```

You can run this example to demo it working too:
``` dart
await NyStorage.delete('coins');

await NyStorage.addToCollection('coins', item: 0);
await NyStorage.addToCollection('coins', item: 1);
await NyStorage.addToCollection('coins', item: 2); // update this
await NyStorage.addToCollection('coins', item: 3);

List<dynamic> coins = await NyStorage.readCollection('coins');
print('before');
for (var coin in coins) {
      print(coin);
}

// update
await NyStorage.updateCollectionByIndex(2, (item) {
      return 200;
}, key: 'coins');

print('\n');

print('after');
coins = await NyStorage.readCollection('coins');
// before
for (var coin in coins) {
      print(coin);
}
```
